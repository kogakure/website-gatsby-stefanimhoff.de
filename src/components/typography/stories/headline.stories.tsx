import React from 'react';

import { Headline } from '..';

export default {
  title: 'Typography/Headline',
};

export const Default = () => <Headline>Headline</Headline>;

export const NoMargin = () => <Headline noMargin>Headline</Headline>;
