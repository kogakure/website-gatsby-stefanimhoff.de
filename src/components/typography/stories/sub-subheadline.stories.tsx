import React from 'react';

import { SubSubheadline } from '..';

export default {
  title: 'Typography/SubSubheadline',
};

export const Default = () => <SubSubheadline>SubSubheadline</SubSubheadline>;

export const NoMargin = () => (
  <SubSubheadline noMargin>SubSubheadline</SubSubheadline>
);
