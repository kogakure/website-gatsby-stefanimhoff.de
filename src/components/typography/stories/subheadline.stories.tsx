import React from 'react';

import { Subheadline } from '..';

export default {
  title: 'Typography/Subheadline',
};

export const Default = () => <Subheadline>Subheadline</Subheadline>;

export const NoMargin = () => <Subheadline noMargin>Subheadline</Subheadline>;
