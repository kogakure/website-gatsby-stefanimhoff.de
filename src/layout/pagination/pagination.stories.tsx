import React from 'react';

import { Pagination, LinkWithPaginationProps } from '.';

export default {
  title: 'Layout/Pagination',
};

const leftProps: LinkWithPaginationProps = {
  text: 'Previous',
  to: '/previous/',
  variant: 'left',
};

const rightProps: LinkWithPaginationProps = {
  text: 'Next',
  to: '/next/',
  variant: 'right',
};

export const Default = () => <Pagination to="/" variant="right" />;

export const Both = () => (
  <>
    <Pagination {...leftProps} />
    <Pagination {...rightProps} />
  </>
);

export const Left = () => <Pagination {...leftProps} />;

export const Right = () => <Pagination {...rightProps} />;
