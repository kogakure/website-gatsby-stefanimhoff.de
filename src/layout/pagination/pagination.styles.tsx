import styled, { css } from 'styled-components';
import { Link as GatsbyLink } from 'gatsby';

import { ArrowLeftS, ArrowRightS } from '../../components/icons';

import { PaginationProps } from '.';

export const variants = {
  left: css`
    left: 1rem;
  `,
  right: css`
    right: 1rem;
  `,
};

const link = css`
  border-radius: 50%;
  position: fixed;
  top: calc(50% - 20px);
`;

const pagination = css`
  align-items: center;
  background-color: var(--colorForegroundFeather);
  border-radius: 50%;
  display: flex;
  height: 40px;
  justify-content: center;
  width: 40px;

  a:hover &,
  a:focus & {
    background-color: var(--colorForegroundSoft);
  }
`;

const Link = styled(GatsbyLink)<PaginationProps>`
  ${link};
  ${({ variant }) => variant && variants[variant]}
`;

const Pagination = styled.div<PaginationProps>`
  ${pagination};
`;

const LeftArrow = styled(ArrowLeftS)`
  fill: var(--colorForeground);
`;

const RightArrow = styled(ArrowRightS)`
  fill: var(--colorForeground);
`;

export const Styled = {
  LeftArrow,
  Link,
  Pagination,
  RightArrow,
};
