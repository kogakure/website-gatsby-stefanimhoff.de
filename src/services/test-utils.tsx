import React from 'react';
import { render, RenderOptions, RenderResult } from '@testing-library/react';

import { EmojifyProvider, LocalizedDateProvider } from '../contexts';

const customRender = (
  node: React.ReactElement,
  options?: RenderOptions
): RenderResult => {
  const rendered = render(
    <EmojifyProvider>
      <LocalizedDateProvider>{node}</LocalizedDateProvider>
    </EmojifyProvider>,
    options
  );

  return {
    ...rendered,
    rerender: (ui: React.ReactElement, rerenderOptions?: RenderOptions) =>
      customRender(ui, { container: rendered.container, ...options }),
  };
};

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
