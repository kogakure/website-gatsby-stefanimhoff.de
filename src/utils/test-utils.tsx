import React from 'react';
import { render, RenderOptions, RenderResult } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

import { EmojifyProvider, LocalizedDateProvider } from '../contexts';
import { ThemeType } from '../typings/Theme';
import { lightTheme } from '../themes';

interface CustomTheme {
  theme: ThemeType;
}

type RenderOptionsWithTheme = RenderOptions & CustomTheme;

const customRender = (
  node: React.ReactElement,
  options: RenderOptionsWithTheme = { theme: lightTheme }
): RenderResult => {
  const { theme } = options;
  const rendered = render(
    <ThemeProvider theme={theme}>
      <EmojifyProvider>
        <LocalizedDateProvider>{node}</LocalizedDateProvider>
      </EmojifyProvider>
    </ThemeProvider>,
    options
  );

  return {
    ...rendered,
    rerender: (
      ui: React.ReactElement,
      rerenderOptions?: RenderOptionsWithTheme
    ) => customRender(ui, { container: rendered.container, ...options }),
  };
};

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
