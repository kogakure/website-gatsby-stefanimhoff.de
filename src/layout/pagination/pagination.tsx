import * as React from 'react';
import { GatsbyLinkProps } from 'gatsby';

import { Styled, variants } from './pagination.styles';

export type PaginationVariant = keyof typeof variants;
export type PaginationProps = {
  text?: string;
  variant: PaginationVariant;
};

/* eslint-disable */
export type LinkWithPaginationProps = PaginationProps &
  Pick<GatsbyLinkProps<any>, 'to'>;
/* eslint-enable */

export const Pagination: React.FC<LinkWithPaginationProps> = ({
  text = '',
  to,
  variant,
  ...props
}) => (
  <Styled.Link aria-label={text} title={text} to={to} variant={variant}>
    <Styled.Pagination variant={variant} {...props}>
      {variant === 'left' && <Styled.LeftArrow aria-hidden="true" />}
      {variant === 'right' && <Styled.RightArrow aria-hidden="true" />}
    </Styled.Pagination>
  </Styled.Link>
);
