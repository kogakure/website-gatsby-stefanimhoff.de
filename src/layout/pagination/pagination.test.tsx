import React from 'react';
import { axe } from 'jest-axe';

import { render } from '../../services/test-utils';

import { Default, Both, Left, Right } from './pagination.stories';

describe('Pagination', () => {
  test('renders correctly', () => {
    const { container } = render(<Default />);

    expect(container.firstChild).toMatchSnapshot();
  });

  test('"Both" renders correctly', () => {
    const { container } = render(<Both />);

    expect(container.firstChild).toMatchSnapshot();
  });

  test('"Left" renders correctly', () => {
    const { container } = render(<Left />);

    expect(container.firstChild).toMatchSnapshot();
  });

  test('"Right" renders correctly', () => {
    const { container } = render(<Right />);

    expect(container.firstChild).toMatchSnapshot();
  });

  test('is accessible', async () => {
    const { container } = render(<Both />);
    const results = await axe(container);

    expect(results).toHaveNoViolations();
  });
});
