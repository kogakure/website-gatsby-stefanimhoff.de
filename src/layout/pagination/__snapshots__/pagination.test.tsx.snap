// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination "Both" renders correctly 1`] = `
.c0 {
  border-radius: 50%;
  position: fixed;
  top: calc(50% - 20px);
  left: 1rem;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: var(--colorForegroundFeather);
  border-radius: 50%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 40px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 40px;
}

a:hover .c1,
a:focus .c1 {
  background-color: var(--colorForegroundSoft);
}

.c2 {
  fill: var(--colorForeground);
}

<a
  aria-label="Previous"
  class="c0"
  href="/previous/"
  title="Previous"
  variant="left"
>
  <div
    class="c1"
  >
    <svg
      aria-hidden="true"
      class="c2"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M10.828 12l4.95 4.95-1.414 1.414L8 12l6.364-6.364 1.414 1.414z"
      />
    </svg>
  </div>
</a>
`;

exports[`Pagination "Left" renders correctly 1`] = `
.c0 {
  border-radius: 50%;
  position: fixed;
  top: calc(50% - 20px);
  left: 1rem;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: var(--colorForegroundFeather);
  border-radius: 50%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 40px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 40px;
}

a:hover .c1,
a:focus .c1 {
  background-color: var(--colorForegroundSoft);
}

.c2 {
  fill: var(--colorForeground);
}

<a
  aria-label="Previous"
  class="c0"
  href="/previous/"
  title="Previous"
  variant="left"
>
  <div
    class="c1"
  >
    <svg
      aria-hidden="true"
      class="c2"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M10.828 12l4.95 4.95-1.414 1.414L8 12l6.364-6.364 1.414 1.414z"
      />
    </svg>
  </div>
</a>
`;

exports[`Pagination "Right" renders correctly 1`] = `
.c0 {
  border-radius: 50%;
  position: fixed;
  top: calc(50% - 20px);
  right: 1rem;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: var(--colorForegroundFeather);
  border-radius: 50%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 40px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 40px;
}

a:hover .c1,
a:focus .c1 {
  background-color: var(--colorForegroundSoft);
}

.c2 {
  fill: var(--colorForeground);
}

<a
  aria-label="Next"
  class="c0"
  href="/next/"
  title="Next"
  variant="right"
>
  <div
    class="c1"
  >
    <svg
      aria-hidden="true"
      class="c2"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M13.172 12l-4.95-4.95 1.414-1.414L16 12l-6.364 6.364-1.414-1.414z"
      />
    </svg>
  </div>
</a>
`;

exports[`Pagination renders correctly 1`] = `
.c0 {
  border-radius: 50%;
  position: fixed;
  top: calc(50% - 20px);
  right: 1rem;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: var(--colorForegroundFeather);
  border-radius: 50%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 40px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 40px;
}

a:hover .c1,
a:focus .c1 {
  background-color: var(--colorForegroundSoft);
}

.c2 {
  fill: var(--colorForeground);
}

<a
  aria-label=""
  class="c0"
  href="/"
  title=""
  variant="right"
>
  <div
    class="c1"
  >
    <svg
      aria-hidden="true"
      class="c2"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M13.172 12l-4.95-4.95 1.414-1.414L16 12l-6.364 6.364-1.414-1.414z"
      />
    </svg>
  </div>
</a>
`;
